File: chronicle-web/src/types/spell.ts
export interface Spell {
  id: string
  name: string
  spell_level: string
  school?: string
  description?: string
  duration?: string
  components?: string
  saving_throw?: string
  range?: string
  source?: string
  targets?: string
  casting_time?: string
  ritual: boolean
  isNativeSpell: boolean
  isAlwaysPrepared: boolean
  isPrepared: boolean
  isDomainSpell: boolean
  isFavoriteSpell: boolean
}

export interface PreparedSpell {
  id: string
  spellId: string
  isPrepared: boolean
  isAlwaysPrepared: boolean
  isFavorite: boolean
  isDomainSpell: boolean
}

export interface SpellsPerDay {
  id: string
  spellLevel: number
  maxSlots: number
  usedSlots: number
  date: string
}

export interface SpellCastLog {
  id: string
  spellName?: string
  spellLevel: number
  castTime: string
  castAsRitual: boolean
  success: boolean
  reason?: string
  failedReason?: string
  sessionId: number
}

File: Chronicle.Api/Models/Spell.cs
using Newtonsoft.Json;
using Chronicle.Api.Models.Enums;

namespace Chronicle.Api.Models;

public class Spell
{
    [JsonProperty("name")] public string? Name { get; set; }
    [JsonProperty("spell_level")] public string? SpellLevel { get; set; }
    public string Id { get; set; } = "";
    public School? School { get; set; }
    [JsonProperty("description")] public string? Description { get; set; }
    [JsonProperty("duration")] public string? Duration { get; set; }
    [JsonProperty("components")] public string? Components { get; set; }
    [JsonProperty("saving_throw")] public string? SavingThrow { get; set; }
    [JsonProperty("range")] public string? Range { get; set; }
    [JsonProperty("source")] public string? Source { get; set; }
    [JsonProperty("targets")] public string? Targets { get; set; }
    [JsonProperty("casting_time")] public string? CastingTime { get; set; }
    [JsonProperty("ritual")] public bool Ritual { get; set; }
    public bool IsNativeSpell { get; set; }
    public bool IsAlwaysPrepared { get; set; }
    public bool IsPrepared { get; set; }
    public bool IsDomainSpell { get; set; }
    public bool IsFavoriteSpell { get; set; }
}
